x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/chart_switch.tsx

```diff
@@ -173,28 +177,40 @@ export function ChartSwitch(props: Props) {
     };
   }
 
-  const visualizationTypes = useMemo(
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const visualizationTypeGroups = useMemo(
     () =>
       flyoutOpen &&
-      flatten(
-        Object.values(props.visualizationMap).map((v) =>
-          v.visualizationTypes.map((t) => ({
-            visualizationId: v.id,
-            ...t,
-            icon: t.icon,
-          }))
+      groupBy(
+        flatten(
+          Object.values(props.visualizationMap).map((v) =>
+            v.visualizationTypes.map((t) => ({
+              visualizationId: v.id,
+              ...t,
+              icon: t.icon,
+            }))
+          )
         )
-      ).map((visualizationType) => ({
-        ...visualizationType,
-        selection: getSelection(visualizationType.visualizationId, visualizationType.id),
-      })),
+          .filter(
```

Comment:
Nitpick: moving the `filter` within the `Object.values().map()` loop there's less work